### PR TITLE
Entity viewer top bar showing the feature information

### PR DESCRIPTION
**Pull request by ens-ma7**
## Type

- [ ] Bug fix
- [x] New feature
- [ ] Improvement to existing feature

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-463

## Description
Entity viewer top bar to display the feature information, for now its Gene.

## Views affected
Entity Viewer only

- [ ] I have checked any requirements for Google Analytics
- [ ] I have created any required tests
- [ ] The PR has as its final commit a WASM/JS update commit if any Rust files were updated.

## Possible complications
none

### CODE REVIEW

File: src/ensembl/src/content/app/entity-viewer/EntityViewer.tsx [hunk 1]
```diff
@@ -46,6 +51,9 @@ const EntityViewer = (props: Props) => {
 
   useEffect(() => {
     props.setDataFromUrl(params);
+    if (params.entityId) {
+      props.updateEnsObject(params.entityId);
+    }
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/EntityViewer.tsx]: should this perhaps be the part of how the reducer reacts to the `setDataFromUrl ` action?

File: src/ensembl/src/content/app/entity-viewer/components/entity-viewer-topbar/EntityViewerTopbar.scss [hunk 2]
```diff
@@ -0,0 +1,12 @@
+@import 'src/styles/common';
+
+.container {
+  display: flex;
+  width: 100%;
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/components/entity-viewer-topbar/EntityViewerTopbar.scss]: does this have to be specified? isn't it the default?

File: src/ensembl/src/content/app/entity-viewer/components/entity-viewer-topbar/EntityViewerTopbar.scss [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  align-items: center;
+  position: relative;
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/components/entity-viewer-topbar/EntityViewerTopbar.scss]: why position relative?

**Comment by azangru** [comment]
image

this is an empty file

File: src/ensembl/src/content/app/entity-viewer/components/entity-viewer-topbar/EntityViewerTopbar.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { connect } from 'react-redux';
+
+import ScaleSwitcher from './scale-switcher/ScaleSwitcher';
+import FeatureSummaryStrip from 'src/shared/components/feature-summary-strip/FeatureSummaryStrip';
+
+import { RootState } from 'src/store';
+import { EnsObject } from 'src/shared/state/ens-object/ensObjectTypes';
+import { getEntityViewerActiveEnsObject } from '../../state/general/entityViewerGeneralSelectors';
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/components/entity-viewer-topbar/EntityViewerTopbar.tsx]: When you find yourself going several directories up, it's better import modules by starting from `src`.

File: src/ensembl/src/content/app/entity-viewer/components/entity-viewer-topbar/scale-switcher/ScaleSwitcher.scss [hunk 5]
```diff
@@ -0,0 +1,5 @@
+@import 'src/styles/common';
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/components/entity-viewer-topbar/scale-switcher/ScaleSwitcher.scss]: this import is not used anywhere in the styles file.

File: src/ensembl/src/content/app/entity-viewer/components/entity-viewer-topbar/scale-switcher/ScaleSwitcher.scss [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+
+.container {
+  width: 100%;
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/components/entity-viewer-topbar/scale-switcher/ScaleSwitcher.scss]: this class is applied to a div; div has display: block by default, which has width: 100% by default.

File: src/ensembl/src/content/app/entity-viewer/state/general/entityViewerGeneralActions.ts [hunk 7]
```diff
@@ -70,3 +74,30 @@ export const changeActiveGenomeId: ActionCreator<ThunkAction<
...
+      return;
+    }
+    const currentActiveEnsObjectIds = getEntityViewerActiveEnsObjectIds(state);
+    const updatedActiveEnsObjectIds = {
+      ...currentActiveEnsObjectIds,
+      [activeGenomeId]: activeEnsObjectId
+    };
+
+    dispatch(updateEntityViewerActiveEnsObjectIds(updatedActiveEnsObjectIds));
+    dispatch(fetchEnsObject(activeEnsObjectId));
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/state/general/entityViewerGeneralActions.ts]: fetching should be conditional on whether we have already received this data

File: src/ensembl/src/content/app/entity-viewer/components/entity-viewer-topbar/scale-switcher/ScaleSwitcher.scss (hunk 6 above)
Comment (ens-ma7) [reply src/ensembl/src/content/app/entity-viewer/components/entity-viewer-topbar/scale-switcher/ScaleSwitcher.scss]: Yep agreed. I just realised it was partly finished code cleanup that i have pushed. But keep the comment coming i am pretty much there with the cleanup.

File: src/ensembl/src/content/app/entity-viewer/components/entity-viewer-topbar/EntityViewerTopbar.scss [hunk 8]
```diff
@@ -0,0 +1,12 @@
+@import 'src/styles/common';
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/components/entity-viewer-topbar/EntityViewerTopbar.scss]: import not used in the file

**Comment by ens-ma7** [comment]
@azangru implemented the suggested changes, please have a look and let me know. THanks

**Comment by azangru** [comment]
This PR raises an important question about how we represent entities in entity viewer.

EnsObjects was our idea of entities in the genome browser page (they represent what we call a "focus object"). We did wonder whether they will exist in other parts of the site, and for a bit thought that they might, but it’s becoming increasingly clear that they are not the same as entities in the entity viewer.

As soon as the backend api prototype becomes available for EntityViewer, there will be no ensObjects. We can start preparing for that now, by not admitting ensObjects into the EntityViewer state (or components).

**Comment by ens-ma7** [comment]
@azangru i am aware of this and there will be a separate ticket to get this sorted (i haven't created yet) and will add the bit about removing ensObj from the state. For now i just wanted to get it working and also it did help me to learn a bit about how state work in react. Can we please leave it for now and i will add your comment in the jira ticket that i create to fetch the data.
Thanks

**Comment by ens-ma7** [comment]
Just to clarify after discussion with @azangru @imransl we decided to leave ensObject in the state as we might need some of the parameters like stable_id, genome_id to work on other parts of the entity viewer until the backend API is ready.

**Pull request merged**